expression/expression.go

```diff
@@ -126,10 +127,11 @@ func (s Schema) FindColumn(astCol *ast.ColumnName) (*Column, error) {
 		if (dbName.L == "" || dbName.L == col.DBName.L) &&
 			(tblName.L == "" || tblName.L == col.TblName.L) &&
 			(colName.L == col.ColName.L) {
-			if idx != -1 {
+			if idx != -1 && !col.Auxiliary {
```

Comment:
For, SELECT 1-d as d FROM T having d + 1 < 0 order by d + 1 .
having d + 1 resolve by 1-d as d, order by d + 1 resolve by T.d.
Then the Select fields will become SELECT 1-d as d, t.d(aux) , there are two columns named d.
For having d + 1, it's confused. So if we find a non-aux column named d, the aux one is ignored.